File: 1. 단계별로_풀어보기/30~39/30. 투 포인터/2. 2470.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const N = +inputArr[0]
const liquidArr = inputArr[1].trim().split(' ').map(Number)

// 우선 오름차순으로 정렬한다.
liquidArr.sort((a, b) => a - b)

// 결과 저장할 변수
let resultSum = 2000000001
let resultLeft = 0
let resultRight = 0

// 양 끝 인덱스
let left = 0
let right = N - 1

while (left < right) {
  let tempSum = liquidArr[left] + liquidArr[right]

  if (Math.abs(tempSum) < resultSum) {
    resultSum = Math.abs(tempSum)
    resultLeft = liquidArr[left]
    resultRight = liquidArr[right]
  }

  if (0 < tempSum) {
    right--
  } // ::
  else {
    left++
  }
}

process.stdout.write(`${resultLeft} ${resultRight}`)

File: 1. 단계별로_풀어보기/30~39/30. 투 포인터/2. 2470.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

N = int(input())
liquidArr = list(map(int, input().split()))

liquidArr.sort()

result = 2 * (10 ** 9) + 1
small = 0
big = 0

left = 0
right = N - 1

while left < right:
  tempSum = liquidArr[left] + liquidArr[right]

  if abs(tempSum) < result:
    result = abs(tempSum)
    small = liquidArr[left]
    big = liquidArr[right]
  
  if tempSum > 0:
    right -= 1
  else:
    left += 1

output(f"{small} {big}")